### PR TITLE
Add from address option in make release

### LINKED ISSUE #5516: make-release should take the deployer address as an argument

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

- Adds `--from` address option to make release

### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

_An explanation of how the changes were tested or an explanation as to why they don't need to be._

### Related issues

- Fixes #5516

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### CODE REVIEW

File: packages/protocol/scripts/bash/make-release.sh [hunk 1]
```diff
@@ -11,25 +11,25 @@ set -euo pipefail
...
-while getopts 'b:n:p:i:r:dt:' flag; do
+while getopts 'b:n:p:i:r:df:' flag; do
   case "${flag}" in
     b) BRANCH="${OPTARG}" ;;
     n) NETWORK="${OPTARG}" ;;
     p) PROPOSAL="${OPTARG}" ;;
     i) INITIALIZE_DATA="${OPTARG}" ;;
     r) REPORT="${OPTARG}" ;;
     d) DRYRUN="--dry_run" ;;
-    t) TRUFFLE_OVERRIDE="${OPTARG}" ;;
```
Comment (m-chrzan) [inline packages/protocol/scripts/bash/make-release.sh:32]: Do we not need the `TRUFFLE_OVERRIDE` anymore?

File: packages/protocol/scripts/truffle/make-release.ts [hunk 2]
```diff
@@ -88,26 +88,20 @@ const deployImplementation = async (
...
 
-const deployProxy = async (contractName: string, addresses: ContractAddresses, dryRun: boolean) => {
-  // Explicitly forbid upgrading to a new Governance proxy contract.
-  // Upgrading to a new Governance proxy contract would require ownership of all
-  // contracts to be moved to the new governance contract, possibly including contracts
-  // deployed in this script.
-  // Because this depends on ordering (i.e. was the new GovernanceProxy deployed
-  // before or after other contracts in this script?), and that ordering is not being
-  // checked, fail if there are storage incompatible changes to Governance.
-  if (contractName === 'Governance') {
```
Comment (m-chrzan) [inline packages/protocol/scripts/truffle/make-release.ts]: Do we not need this check anymore?

Comment (m-chrzan) [reply packages/protocol/scripts/truffle/make-release.ts]: Ah, see it's been moved lower. Should it remain here to keep the Proxy deployment logic in the `deployProxy` function?

File: packages/protocol/scripts/bash/make-release.sh (hunk 1 above)
Comment (yorhodes) [reply packages/protocol/scripts/bash/make-release.sh:32]: apparently it never worked

File: packages/protocol/scripts/truffle/make-release.ts (hunk 2 above)
Comment (yorhodes) [reply packages/protocol/scripts/truffle/make-release.ts]: for some reason I thought this was easier than passing around the from address
will revert

**Review by nambrot — APPROVED** [review]

**Pull request merged**